### PR TITLE
Fix multiple escape sequences

### PR DESCRIPTION
**Pull request by Flakebi**
If escape sequences follow straight after each other, there is no
delimiter in-between.
In such a case, parsing previously stopped and did not find any
typos further in the file.

Also change the c-escape parser to take either \\+\<single char> or
\\+\<char sequence delimited by space or similar>.

Also update trycmd, so failures actually show up in CI.

### CODE REVIEW

File: tests/cmd/stdin-failure2.toml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+bin.name = "typos"
+args = "-"
+status.code = 2
```
Comment (epage) [inline tests/cmd/stdin-failure2.toml:3]: Should we make this case `--write-changes` so we can validate the `2` is for the reason we think?

Comment (Flakebi) [reply tests/cmd/stdin-failure2.toml]: Good idea

**Review by epage — COMMENTED** [review]
Sorry, I thought these were published

File: crates/typos/src/tokens.rs [hunk 2]
```diff
@@ -410,7 +410,12 @@ mod parser {
         <T as nom::InputTakeAtPosition>::Item: AsChar + Copy,
         <T as nom::InputIter>::Item: AsChar + Copy,
     {
-        preceded(take_while1(is_escape), take_while(is_xid_continue))(input)
+        // Match either a multi char escape sequence like \x01 until a separator
+        // or a single char escape like \<newline> or \<space>.
+        preceded(
+            take_while1(is_escape),
+            alt((take_while1(is_xid_continue), take(1usize))),
+        )(input)
```
[the remark this answers is not in the document]
Comment (Flakebi) [reply crates/typos/src/tokens.rs]: That makes a lot of sense, I added that as a comment.

**Review by epage — APPROVED** [review]
Thanks!

**Comment by epage** [comment]
1.3.0 is now released!

**Comment by Flakebi** [comment]
Thank you!

**Pull request merged**